python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -646,6 +649,96 @@ async def main():
     compiled_dag.teardown()
 
 
+class TestMultiChannel:
+    def test_multi_channel_one_actor(self, ray_start_regular_shared):
+        """
+        In this test, there are three 'inc' tasks on the same Ray actor, chained
+        together. Therefore, the DAG will look like this:
+
+        Driver -> a.inc -> a.inc -> a.inc -> Driver
+
+        All communication between the driver and the actor will be done through remote
+        channels, i.e., shared memory channels. All communication between the actor
+        tasks will be conducted through local channels, i.e., IntraProcessChannel in
+        this case.
+
+        To elaborate, all output channels of the actor DAG nodes will be MultiChannel,
+        and the first two will have a local channel, while the last one will have a
+        remote channel.
+        """
+        a = Actor.remote(0)
+        with InputNode() as inp:
+            dag = a.inc.bind(inp)
+            dag = a.inc.bind(dag)
+            dag = a.inc.bind(dag)
+
+        compiled_dag = dag.experimental_compile()
+        assert len(compiled_dag.actor_to_tasks) == 1
+
+        num_local_channels = 0
+        num_remote_channels = 0
+        for tasks in compiled_dag.actor_to_tasks.values():
+            assert len(tasks) == 3
+            for task in tasks:
+                assert isinstance(task.output_channel, MultiChannel)
```

Comment:
Hmm ideally unit tests should not assume a certain implementation or that certain internal fields are available to call. Can you perhaps do something like a mocked approach where we check how many Channel vs IntraProcessChannel constructors or methods are called by the end of the test? You can check out the `TracedChannel` test util that I added in this [open PR](https://github.com/ray-project/ray/pull/45677/files#diff-4cc824d545e027fce84524333a8f839fd136bc8bf996efffaa7039c9acac4ee4R126) for an example of this.